packages/components/src/style-provider/index.js

```diff
@@ -0,0 +1,20 @@
+/**
+ * External dependencies
+ */
+import { CacheProvider } from '@emotion/core';
+import createCache from '@emotion/cache';
+import memoize from 'memize';
+
+const memoizedCreateCacheWithContainer = memoize( ( container ) => {
+	return createCache( { container } );
+} );
+
+export default function StyleProvider( { children, iframeDocument } ) {
+	if ( ! iframeDocument ) {
+		return null;
+	}
+
+	const cache = memoizedCreateCacheWithContainer( iframeDocument.head );
+
+	return <CacheProvider value={ cache }>{ children }</CacheProvider>;
```

Comment:
I didn't know this existed. It's great that emotion provides this `CacheProvider` and consumes it.